python/lsst/pipe/tasks/loadReferenceCatalog.py

```diff
@@ -0,0 +1,380 @@
+
+        return self._formatCatalog(refCat, filterList)
+
+    def _formatCatalog(self, refCat, filterList):
+        """Format a reference afw table into the final format.
+
+        This method applies reference selections and color terms as specified
+        by the config.
+
+        Parameters
+        ----------
+        refCat : `lsst.afw.table.SourceCatalog`
+            Reference catalog in afw format.
+        filterList : `list` [`str`]
+            List of camera physicalFilter names to apply color terms.
+
+        Returns
+        -------
+        refCat : `numpy.ndarray`
+            Reference catalog.
+        """
+        if self.config.doReferenceSelection:
+            goodSources = self.referenceSelector.selectSources(refCat)
+            selected = goodSources.selected
+        else:
+            selected = np.ones(len(refCat), dtype=bool)
+
+        npRefCat = np.zeros(np.sum(selected), dtype=[('ra', 'f8'),
+                                                     ('dec', 'f8'),
+                                                     ('refMag', 'f4', (len(filterList), )),
+                                                     ('refMagErr', 'f4', (len(filterList), ))])
+
+        if npRefCat.size == 0:
+            # Return an empty catalog if we don't have any selected sources.
+            return npRefCat
+
+        # Natively "coord_ra" and "coord_dec" are stored in radians.
+        # Doing this as an array rather than by row with the coord access is
+        # approximately 600x faster.
+        npRefCat['ra'] = np.rad2deg(refCat['coord_ra'][selected])
+        npRefCat['dec'] = np.rad2deg(refCat['coord_dec'][selected])
+
+        # Default (unset) values are 99.0
+        npRefCat['refMag'][:, :] = 99.0
+        npRefCat['refMagErr'][:, :] = 99.0
+
+        if self.config.doApplyColorTerms:
+            if isinstance(self.refObjLoader, ReferenceObjectLoader):
+                # Gen3
+                refCatName = self.refObjLoader.config.value.ref_dataset_name
+            else:
+                # Gen2
+                refCatName = self.refObjLoader.ref_dataset_name
+
+            for i, (filterName, fluxField) in enumerate(zip(self._fluxFilters, self._fluxFields)):
+                if fluxField is None:
+                    # There is no matching reference band.
+                    # This will leave the column filled with 99s
+                    continue
+                self.log.debug(f"Applying color terms for filtername={filterName}")
+
+                colorterm = self.config.colorterms.getColorterm(filterName, refCatName, doRaise=True)
+
+                refMag, refMagErr = colorterm.getCorrectedMagnitudes(refCat)
+
+                # nan_to_num replaces nans with zeros, and this ensures that we select
+                # magnitudes that both filter out nans and are not very large (corresponding
+                # to very small fluxes), as "99" is a commen sentinel for illegal magnitudes.
+
```

Comment:
I think it might make more sense for this newline to be removed, so the above comment block is clearly attached to the code it annotates.